main.tf

```diff
@@ -161,6 +218,27 @@ resource "metal_port" "esxi_hosts" {
   }
 }
 
+data "template_file" "vars_file" {
```

Comment:
The `templatefile` function is preferred:

https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file

(this change is replacing templatefile with a datasource template_file)